### ISSUE TITLE
Create a PR on the main mitsuba repository with the code changes to add the photon_emitter on this fork

### ISSUE THREAD

**Issue by andrewgait** [UoMResearchIT/mitsuba3-manchester#16]
This may be more relevant in a few sprints from now, but I thought it worth creating a ticket to discuss what to do with the new code on this fork; ideally, I imagine we want to merge our new emitter into the main mitsuba3 repository.

To that end, the process will be to add the relevant code that we want to merge into the main mitsuba3 repository to the `main` branch of this fork and then create a pull request from there; and keep anything we don't want to merge into the main mitsuba3 repository (e.g. the performance & scaling tests we've run) on the `dev` branch here (or indeed, on any other branch on this fork should that be necessary).  If desired we could make a tutorial about the photon_emitter, but that would be worth its own ticket.

Below is a list of work that will need to be done before this can happen, which we can add to as necessary.

- [x] #13 
- [x] #9 
- [x] #8 
- [x] #7 
- [x] #4 
- [ ] #23 
- [ ] #25 
- [ ] https://github.com/UoMResearchIT/mitsuba3-manchester/issues/16
- [ ] https://github.com/UoMResearchIT/mitsuba3-manchester/issues/28
- [ ] https://github.com/UoMResearchIT/mitsuba3-manchester/issues/29
